src/main/java/frc/robot/commands/CDSBallManagementCommand.java

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.CDSSubsystem;
+import frc.robot.subsystems.IntakeSubsystem;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import frc.robot.Constants;
+
+
+public class CDSBallManagementCommand extends CommandBase {
+  /** Creates a new CDSBallManagementCommand. */
+  private final CDSSubsystem CDSSubsystem;
+  private final IntakeSubsystem intakeSubsystem;
+
+  private boolean runningCDS = false;
+  private boolean ejectRunning = false;
+  private int setpointIndex;
+
+  private int cycleCount = 0;
+  private int msDelay = 2000;
+
+  public CDSBallManagementCommand(CDSSubsystem mCDSSubsystem, IntakeSubsystem mIntakeSubsystem) {
+    // Use addRequirements() here to declare subsystem dependencies.
+    addRequirements(mCDSSubsystem);
+    addRequirements(mIntakeSubsystem);
+
+    CDSSubsystem = mCDSSubsystem;
+    intakeSubsystem = mIntakeSubsystem;
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {}
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    // Run eject before auto advance
+    int ballCount = (int) SmartDashboard.getNumber("Ball Count", 0);
+    boolean[] sensorStatus = CDSSubsystem.getSensorStatus();
+
+    SmartDashboard.putBoolean("Front sensor status", sensorStatus[2]);
+    SmartDashboard.putBoolean("Middle Sensor Status", sensorStatus[1]);
+    SmartDashboard.putBoolean("Back Sensor Status", sensorStatus[0]);
+
+
+    if (!ejectRunning) {
```

Comment:
Good work on figuring this command out. At some point later we can go over how to write this as a 'state machine'.

Reply:
Sounds good!